### ISSUE TITLE
Missing building blocks of SIGMA

### ISSUE THREAD

**Issue by LemonAndRabbit** [stonne-simulator/stonne#13]
Hi Francisco Muñoz-Martínez,

I have read your awesome paper and now I plan to simulate SIGMA on STONNE. After reading your codes, I found that in your [SIGMA configuration file](https://github.com/stonne-simulator/stonne/blob/master/simulation_files/sigma_128mses_128_bw.cfg):

```cfg
[MSNetwork]
ms_size=128
[ReduceNetwork]
type="ASNETWORK"
[SDMemory]
dn_bw=128
rn_bw=128
controller_type="SIGMA_SPARSE_GEMM"
```

According to the codes, I think it used the default *trees-like* DSNetworkTop Distribution Network and *Augmented Reduction Tree(ART)* as Reduction Network. But in the [SIGMA paper](https://cpn-us-w2.wpmucdn.com/sites.gatech.edu/dist/c/332/files/2020/01/sigma_hpca2020.pdf), they used *Benes Topology* Distribution Network and *Forwarding Adder Network(FAN)* Reduction Network.

I wonder if the two building blocks of SIGMA are implemented in the current repo and whether the current configuration is capable to simulate the correct behaviour(timing, energy consumed and the on-chip area) of SIGMA?

Thanks a lot.

**Comment by wangjiayue0303** [comment]
Hi. I have the same confusion about the implementation of SIGMA. It seems that the RNs and DNs provided in stonne do not match Figure 3 in the paper. I wonder if I misunderstood some parts. Could you please provide some help?

**Comment by francisco-munoz** [comment]
Dear all,

In terms of performance, both FEN and ART networks are mathematically equivalent. So there was no reason to model a different network to get exactly the same performance behaviour. 

The difference between the networks is mostly on the energy/area consumption side. For these we used different numbers in the paper by applying synthesis to the hardware designs. 
We updated the energy tables of STONNE using these numbers and calculated the energy consumption of each case. That’s why I’m the paper we mention both networks when actually, functionally speaking, we only model one of them.

best,
Francisco
